tests/capture/test_templates.py

```diff
@@ -190,6 +190,15 @@ def enable_disable_plxpr():
 ]
 
 
+@pytest.fixture(scope="function", autouse=True)
+def capture_warnings(recwarn):
+    yield
+    if len(recwarn) > 0:
+        for w in recwarn:
+            assert isinstance(w.message, qml.PennyLaneDeprecationWarning)
+            assert "BasisStatePreparation is deprecated" in str(w.message)
+
+
```

Comment:
You can use that same `filterwarnings` marker here. You can put it at the top of the file as
```python
pytestmark = pytest.mark.filterwarnings("ignore:BasisStatePreparation is deprecated:pennylane.PennyLaneDeprecationWarning")
```